lib/vdsm/storage/blockVolume.py

```diff
@@ -70,6 +70,9 @@ def is_block(cls):
     def chunked(self):
         return self.getFormat() == sc.COW_FORMAT
 
+    def can_reduce(self):
+        return self.chunked() and self.getType() == sc.SPARSE_VOL
```

Comment:
This is fine for cold merge, where we use BlockVolumeManifest object. But in Volume.reduce, we use:
https://github.com/oVirt/vdsm/blob/bec5e7c94d25d02cadb3e9c9a481a6963151211c/lib/vdsm/storage/blockSD.py#L1600

Which uses a BlockVolume object (using a BlockVolumeManifest).

Why do we have both BlockVolume and BlockVolumeManifest? this is a result of cancelled project to remove the SPM in ovirt 4.0. The plan was to extract BlockVolumeManifest that will be used for new code path in ovirt 4.0, and remove BlockVolume in later release. Sadly this project was canceled in the middle, leaving us with worse code.

So what we have do now, is create wrappers for code that need to be accessed from both BlockVolume and BlockVolumeManifast. Most of the wrappers are in sd.py on the StorageDomain super class. See StorageDomain.supports_bitmap_operations() for example wrapper.

We have very bad tests for the wrapers in sdm_indirection_test.py, don't waste your time adding tests there, we need to remove these tests.

Reply:
I hope I got it right. Not sure if you were actually suggesting me to make the wrapper in the `StorageDomain`, or just take it as an example. But to me, it looked better fit to make the wrapper in the `Volume` class.

Made it in a separate commit to make it easier to spot and fix if needed :)